samples/javascript_nodejs/49.qnamaker-all-features/dialogs/qnamakerBaseDialog.js

```diff
@@ -27,10 +27,11 @@ class QnAMakerBaseDialog extends QnAMakerDialog {
      * Core logic of QnA Maker dialog.
      * @param {QnAMaker} qnaService A QnAMaker service object.
      */
-    constructor(knowledgebaseId, authkey, host) {
-        var noAnswer = ActivityFactory.fromObject(DefaultNoAnswer);
+    constructor(knowledgebaseId, authkey, host, defaultAnswer) {
+
+        var defaultAnswerActivity = (!defaultAnswer || !defaultAnswer.trim()) ? ActivityFactory.fromObject(DefaultAnswer) :  ActivityFactory.fromObject(defaultAnswer);
```

Comment:
Could this not be written using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)? Something like:

var defaultAnswerActivity = ActivityFactory.fromObject(defaultAnswer?.trim() ? defaultAnswer : DefaultAnswer);